File: interpreter/java/JobManager.java
package Interpreter;

import java.util.ArrayList;
import java.util.List;

class JobManager {
    public List<IJobExpression> createExpressionTree(String formula) {
        List<IJobExpression> tree = new ArrayList<>();

        for (char role : formula.toCharArray()) {
            switch (role) {
                case 'G':
                    tree.add(new DeveloperExpression());
                    break;
                case 'M':
                    tree.add(new AccountingExpression());
                    break;
                case 'L':
                    tree.add(new TeamLeadExpression());
                    break;
                default:
                    new Exception("Unexpected role!");
            }
        }
        return tree;
    }

    public void runExpression(Context context) {
        for (IJobExpression expression : createExpressionTree(context.formula)) {
            expression.interpret(context);
        }
    }
}

File: interpreter/kotlin/JobManager.kt
package Interpreter

import java.util.ArrayList

internal class JobManager {
    fun createExpressionTree(formula: String): List<IJobExpression> {
        val tree = ArrayList<IJobExpression>()

        for (role in formula.toCharArray()) {
            when (role) {
                'G' -> tree.add(DeveloperExpression())
                'M' -> tree.add(AccountingExpression())
                'L' -> tree.add(TeamLeadExpression())
                else -> Exception("Unexpected role!")
            }
        }
        return tree
    }

    fun runExpression(context: Context) {
        for (expression in createExpressionTree(context.formula!!)) {
            expression.interpret(context)
        }
    }
}